### PR TITLE
Add support for Electron based browser for YomiNinja

### PR DESCRIPTION
**Pull request by entropyofchaos**
[YomiNinja](https://github.com/matt-m-o/YomiNinja) is a project that uses OCR on your screen to scan applications like video games so that you can more easily try to translate them while playing to learn Japanese. On the backend, it makes use of Electron to support chrome extensions. Specifically [electron-chrome-extensions](https://github.com/ramboxapp/electron-chrome-extensions). It seems that this addon doesn't yet support manifest.json version 3. Up to release 1.15.1 you could load 10ten into the project however I did want to be able to use the latest version of your extension to support the new features like bunpro grammar.

I updated the build scripts to create a Electron compatible version of the extension using an older version of the manifest standard.

### CODE REVIEW

**Comment by birtles** [comment]
Thanks for doing this. I think the other thing that would be worth adding here is changes to (`build-assets.sh`](https://github.com/birchill/10ten-ja-reader/blob/b7fc2da09ccf53874715efc786a46cdfd0c3f884/scripts/build-assets.sh#L50-L52) so that the Electron package is included in the GitHub release and you can download it from there.

Also, I don't think we need to the `start:chrome-electron` script since that's only really useful for hot-reloading which presumably the electron add-on doesn't support.

**Comment by birtles** [comment]
Come to think of it, I was actually hoping to drop MV2 support after updating Firefox, Safari, and Thunderbird to MV3. I don't suppose there's anyone working on getting `electron-chrome-extensions` updated?

**Comment by entropyofchaos** [comment]
> Thanks for doing this. I think the other thing that would be worth adding here is changes to [`build-assets.sh`](https://github.com/birchill/10ten-ja-reader/blob/b7fc2da09ccf53874715efc786a46cdfd0c3f884/scripts/build-assets.sh#L50-L52) so that the Electron package is included in the GitHub release and you can download it from there.
> 
> Also, I don't think we need to the `start:chrome-electron` script since that's only really useful for hot-reloading which presumably the electron add-on doesn't support.

Changes made. Honestly Javascript isn't my background so I was just trying to do my best to follow how the other distributions did it and just get the silly thing working with YomiNinja since I liked their project. 

> Come to think of it, I was actually hoping to drop MV2 support after updating Firefox, Safari, and Thunderbird to MV3. I don't suppose there's anyone working on getting `electron-chrome-extensions` updated?

I honestly have no clue. I assume that's well beyond my skills as they are trying to emulate how chrome extensions work in chrome. But leave the choice to you guys as what you want to do. I wish I had more knowledge on browser technologies but I always have to balance my time, learn new tech or learn Japanese. 😄

File: scripts/build-assets.sh [hunk 1]
```diff
@@ -51,6 +51,10 @@ mv dist-firefox-package/10ten_japanese_reader_rikaichamp_-${version}.zip release
 RELEASE_BUILD=1 yarn package:chrome
 mv dist-chrome-package/10ten_japanese_reader_rikaichamp_-${version}.zip release-assets/10ten-ja-reader-${version}-chrome.zip
 
+# Chrome-Electron package
+RELEASE_BUILD=1 yarn package:chrome-electron
+mv dist-chrome-electron-package/10ten_japanese_reader_rikaichamp_-${version}.zip release-assets/10ten-ja-reader-${version}-chrome.zip
```
Comment (birtles) [inline scripts/build-assets.sh:56]:
```suggestion
mv dist-chrome-electron-package/10ten_japanese_reader_rikaichamp_-${version}.zip release-assets/10ten-ja-reader-${version}-chrome-electron.zip
```

**Comment by birtles** [comment]
> I honestly have no clue. I assume that's well beyond my skills as they are trying to emulate how chrome extensions work in chrome. But leave the choice to you guys as what you want to do. I wish I had more knowledge on browser technologies but I always have to balance my time, learn new tech or learn Japanese. 😄

Yes, good call!

I'm happy to add this for now, but there might come a time where we need to drop it if supporting MV2 causes too much maintenance hassle. But hopefully electron-chrome-extensions is updated before then and we can keep this.

**Comment by entropyofchaos** [comment]
> > I'm happy to add this for now, but there might come a time where we need to drop it if supporting MV2 causes too much maintenance hassle. But hopefully electron-chrome-extensions is updated before then and we can keep this.
> 
> I'm working on this if you wanted to track progress: [samuelmaddock/electron-browser-shell#38](https://github.com/samuelmaddock/electron-browser-shell/issues/38)
> 
> I was curious how YomiNinja worked and ended up here :) Thanks for maintaining backwards compatibility in the interim! 🙇

I personally was just trying to use the app and was sad when I saw the new new features of 10ten were missing. It is an app that basically uses OCR to scan a video game screen while you play. Then it uses Chrome extensions to read the text like 10ten. It's pretty helpful for learning Japanese. I've been using it recently to play Ni no Kuni. 

Happy to see V3 support is being worked on. Thanks for your contribution.

**Comment by birtles** [comment]
I might have misunderstood the comment but there should be an electron zip file as part of the latest release so hopefully you can get the latest features.

**Pull request merged**